constraints.txt

```diff
@@ -7,58 +7,73 @@
 # need to submit a PR to pyca/cryptography, since they rely on it.
 
 alabaster==0.7.12
+async-generator==1.10
 attrs==21.2.0
 Babel==2.9.1
-backports.entry-points-selectable==1.1.1
-certifi==2021.10.8
-cffi==1.15.0
-charset-normalizer==2.0.7
-coverage==6.1.1
-cryptography==35.0.0
-distlib==0.3.3
-docutils==0.17.1
-filelock==3.3.2
-flake8==4.0.1
-idna==3.3
+backports.entry-points-selectable==1.1.0
+certifi==2021.5.30
+cffi==1.14.6
+charset-normalizer==2.0.4
+click==8.0.1
+coverage==5.5
+cryptography==3.4.8
+dataclasses-json==0.5.5
+distlib==0.3.2
+docutils==0.17
+filelock==3.0.12
+flake8==3.9.2
+idna==3.2
 imagesize==1.2.0
-importlib-metadata==4.2.0
-importlib-resources==5.4.0
 iniconfig==1.1.1
-Jinja2==3.0.2
+Jinja2==3.0.1
+libcst==0.3.20
 MarkupSafe==2.0.1
+marshmallow==3.13.0
+marshmallow-enum==1.5.1
 mccabe==0.6.1
-mypy==0.812
+mypy==0.910
 mypy-extensions==0.4.3
-packaging==21.2
-platformdirs==2.4.0
+packaging==21.0
+platformdirs==2.3.0
 pluggy==1.0.0
-py==1.11.0
-pycodestyle==2.8.0
-pycparser==2.21
-pyflakes==2.4.0
+psutil==5.8.0
+py==1.10.0
+pycodestyle==2.7.0
+pycparser==2.20
+pyflakes==2.3.1
 Pygments==2.10.0
-pyOpenSSL==21.0.0
+pyOpenSSL==20.0.1
 pyparsing==2.4.7
+pyre-check==0.9.3
+pyre-extensions==0.0.21
 pytest==6.2.5
-pytest-cov==3.0.0
+pytest-cov==2.12.1
 pytest-flake8==1.0.7
-pytz==2021.3
+pytz==2021.1
+pywatchman==1.4.1
+PyYAML==5.4.1
 requests==2.26.0
 six==1.16.0
 snowballstemmer==2.1.0
-Sphinx==4.2.0
+Sphinx==4.1.2
 sphinx-rtd-theme==1.0.0
 sphinxcontrib-applehelp==1.0.2
 sphinxcontrib-devhelp==1.0.2
 sphinxcontrib-htmlhelp==2.0.0
 sphinxcontrib-jsmath==1.0.1
 sphinxcontrib-qthelp==1.0.3
 sphinxcontrib-serializinghtml==1.1.5
+tabulate==0.8.9
 toml==0.10.2
-tomli==1.2.2
-tox==3.24.4
-typed-ast==1.4.3
+tox==3.24.3
+types-cryptography==3.3.5
+types-enum34==1.1.0
+types-ipaddress==1.0.0
+types-pyOpenSSL==20.0.6
+types-pyRFC3339==0.1.2
+types-requests==2.25.6
+types-setuptools==57.0.2
```

Comment:
This works in CI because we currently run `pip install -r constraints.txt`, but `tox -e mypy` fails for me locally.

I think we should fix this by declaring a dependency on these `types-*` packages in `setup.py` like we do in Certbot.

Reply:
Added.